frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java

```diff
@@ -837,6 +850,104 @@ public boolean isSnapshotDisabled() {
+        String fileData = " ";
+        String absoluteFilePath = getCanonicalPath(".") + "/key_file.txt";
+        keyFileLocation = absoluteFilePath;
+        File file = new File(absoluteFilePath);
+        if (!file.createNewFile() && !file.exists()) {
+            return false;
+        }
+        Integer timeToExpiration = 30; // in minutes
+        fileData =
+                "Management Key: "
+                        + generateToken()
+                        + "\n"
+                        + "Inference Key: "
+                        + generateToken()
+                        + " --- Expiration time: "
+                        + Instant.now().plusSeconds(TimeUnit.MINUTES.toSeconds(timeToExpiration))
+                        + "\n";
+        Files.write(Paths.get("key_file.txt"), fileData.getBytes());
+        return true;
+    }
+
+    public List<String> parseFile(File tsTokenFile) {
+        List<String> parsedTokens = new ArrayList<>();
+        try {
+            InputStream stream = Files.newInputStream(tsTokenFile.toPath());
+            byte[] array = new byte[100];
+            stream.read(array);
+            String data = new String(array);
+            String[] arrOfData = data.split("\n", 2);
+            String[] managementArr = arrOfData[0].split(" ", 3);
+            String[] inferenceArr = arrOfData[1].split(" ", 7);
+            parsedTokens.add(managementArr[2]);
+            parsedTokens.add(inferenceArr[2]);
+            String[] expirationArr = inferenceArr[6].split("\n", 2);
+            parsedTokens.add(expirationArr[0]);
+        } catch (IOException | ArrayIndexOutOfBoundsException e) {
+            System.out.println("Unable to read key file or key file has been modified");
+            return null;
+        }
+        return parsedTokens;
+    }
+
+    public void checkTokenAuthorization(FullHttpRequest req, boolean inferenceRequest)
+            throws ModelException {
+        HttpMethod method = req.method();
+        String filePath = keyFileLocation;
+        if (filePath != null) {
+            File tsTokenFile = new File(filePath);
+            if (tsTokenFile.exists()) {
+                List<String> parsedTokens = parseFile(tsTokenFile);
+                String managementToken = parsedTokens.get(0);
+                String inferenceToken = parsedTokens.get(1);
+                Instant expirationTime = Instant.now();
+                try {
+                    expirationTime = Instant.parse(parsedTokens.get(2));
+                } catch (DateTimeException e) {
+                    e.printStackTrace();
+                    System.out.println("{\n\t\"Error\": Key File has been modified \n}\n");
+                }
+                String tokenBearer = req.headers().get("Authorization");
+                if (tokenBearer == null) {
+                    throw new InvalidKeyException("NO TOKEN PROVIDED");
+                }
+                String[] arrOfStr = tokenBearer.split(" ", 2);
```

Comment:
We could make parsing the token from the header more robust using a regex pattern.

Reply:
No longer need to parse the key file.